src/Entity/OgRole.php

```diff
@@ -38,138 +34,135 @@
 class OgRole extends Role implements OgRoleInterface {
 
   /**
-   * @var integer
-   *
-   * The group ID.
-   */
-  protected $group_id;
-
-  /**
-   * The entity type ID of the group.
-   *
-   * @var string
-   */
-  protected $group_type;
-
-  /**
-   * The bundle ID of the group.
-   *
-   * @var string
-   */
-  protected $group_bundle;
-
-  /**
    * Set the ID of the role.
    *
    * @param string $id
    *   The machine name of the role.
    *
-   * @return OgRole
+   * @return $this
```

Comment:
Yes, the `@return $this` pattern is the standard way of doing this in core so PHP CodeSniffer is correct. It is not formally documented in the [API documentation and comment standards](https://www.drupal.org/coding-standards/docs) but there are hundreds of places where this is used in core.

Most importantly, this pattern is used is `\Drupal\user\RoleInterface` which `OgRole` is implementing, so we should definitely follow. Here is an example method declaration on `RoleInterface`:

```
  /**
   * Grant permissions to the role.
   *
   * @param string $permission
   *   The permission to grant.
   *
   * @return $this
   */
  public function grantPermission($permission);
```